tests/data/test_breadth_first_walker.py

```diff
@@ -506,34 +506,35 @@ def test_walk_generation_number_of_walks_per_root_nodes(self):
     def test_fixed_random_seed(self):
 
         g = create_test_graph()
+        _conv = g.node_ids_to_ilocs
```

Comment:
Why is this changing? This PR is a behaviour change?

Reply:
this changes because this test was incorrectly (my bad!) using node ids instead of node ilocs. Previously `_edges_dict.get` would silently return empty lists of neighbours when queired with a string node id,  but `FlatAdjacencyList` only accepts integers as keys and crashes when passed anything else.

I could change `get` to handle strings but it shouldn't ever have to

Reply:
Oh, so these tests were effectively ignoring the strings IDs?

Reply:
yeah they would try to lookup string ids a ilocs, find nothing and return `-1`  nodes, and keep going.